bundles/org.openhab.binding.bambulab/src/main/java/org/openhab/binding/bambulab/internal/PrinterHandler.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.bambulab.internal;
+
+import static org.openhab.binding.bambulab.internal.BambuLabBindingConstants.Channel.*;
+import static org.openhab.core.library.unit.SIUnits.CELSIUS;
+import static org.openhab.core.library.unit.Units.DECIBEL_MILLIWATTS;
+import static org.openhab.core.thing.ThingStatus.*;
+import static org.openhab.core.thing.ThingStatusDetail.CONFIGURATION_ERROR;
+import static org.openhab.core.types.UnDefType.UNDEF;
+import static pl.grzeslowski.jbambuapi.PrinterClient.Channel.LedControlCommand.*;
+import static pl.grzeslowski.jbambuapi.PrinterClient.Channel.LedControlCommand.LedNode.*;
+import static pl.grzeslowski.jbambuapi.PrinterClient.Channel.PushingCommand.defaultPushingCommand;
+import static pl.grzeslowski.jbambuapi.PrinterClientConfig.requiredFields;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import pl.grzeslowski.jbambuapi.PrinterClient;
+import pl.grzeslowski.jbambuapi.PrinterClientConfig;
+import pl.grzeslowski.jbambuapi.PrinterWatcher;
+import pl.grzeslowski.jbambuapi.Report;
+
+/**
+ * The {@link PrinterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class PrinterHandler extends BaseThingHandler implements PrinterWatcher.StateSubscriber {
+    private static final Pattern DBM_PATTERN = Pattern.compile("^(-?\\d+)dBm$");
+    private Logger logger = LoggerFactory.getLogger(PrinterHandler.class);
+
+    private @Nullable PrinterClient client;
+
+    public PrinterHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!LED_CHAMBER_LIGHT_CHANNEL.equals(channelUID.getId())
```

Comment:
The implementation of this method can be improved. Commands for several channels might lead to unexpected `sendCommand`.

Reply:
Can you elaborate?

Reply:
A similar comment was added by jlaur. 

1. Only LED_WORK_LIGHT_CHANNEL will match this condition. If that is intentional, this should be rewritten to 
``` 
if (LED_WORK_LIGHT_CHANNEL.equals(channelUID.getId())) {
```
2. I suspect it should be none of the two channels. But that would meen all except these channels. This is risky as a command send to _any_ channel will trigger the sendCommand call, i can hardly beleave that is a good thing. 

I would opt to only call the sendcommand for a channel it should be called. and log a warning otherwise. Hoep this makes it a bit more clear on what i meant, if not let me know.

Reply:
after remark from @jlaur the method is working correctly. I've added test to make sure it's fine